c++/src/RLE.cc

```diff
@@ -71,14 +71,41 @@ namespace orc {
   }
 
   void RleEncoder::add(const int64_t* data, uint64_t numValues,
-                         const char* notNull) {
+                       const char* notNull) {
     for (uint64_t i = 0; i < numValues; ++i) {
       if (!notNull || notNull[i]) {
         write(data[i]);
       }
     }
   }
 
+  void RleEncoder::add(const int32_t* data, uint64_t numValues,
```

Comment:
Why not use a template function to repeat the code? static_cast does not have any runtime penalty.